### ISSUE TITLE
Create a "blank" default template

### ISSUE THREAD

**Issue by chrisdmacrae** [forestryio/create-static-site#34]
Right now you have to specify `--template hugo|jekyll`. I don't love this for two reasons:

- That's not 0 config ;)
- Pushing Hugo or Jekyll may not the be the right choice

Instead, I propose running `create-static-site <dirname>` creates the bare minimum of a static site -- asset processing & static files.

This could then evolve into a few other "bare-minimum" templates like a react -> toString template.

Thoughts? @ncphillips @budparr @scottgallant ?

**Comment by budparr** [comment]
Those tools have unique directory structures, which would mean you'd manually have to create them, correct?

**Comment by chrisdmacrae** [comment]
@budparr The power of C-S-S is that you can provide a template with any directory structure or files.

In this scenario, I'd just imagine getting a site that looks like:

```
src/
  css/
  js/
  img/
site/
  *.html
dist/
.gitignore
static-scripts.config.js
README.md
```

And my `static-scripts.config.js` would look as follows:

```
/**
 * Configuration file for create-static-site
 * A zero configuration build tool for static sites
 * https://github.com/forestryio/create-static-site/
 */
const {resolve} = require("path")

module.exports = () => ({
    generator: "Generator",
    command: "cp",
    proxy: false,
    args: {
        default: [resolve("site/"), resolve("dist/")],
    },
    directories: {
        src: "src/",
        build: "dist/",
        static: "dist/",
        generator: "site/"
    }
})
```

Just an extremely slim MVP that allows me to prototype static HTML w/ asset processing.

**Comment by budparr** [comment]
At some point you have to specify what tool you're using, why not up front? Perhaps I'm still missing something 🦆 (duck emoji because they had no dunce)

**Comment by chrisdmacrae** [comment]
I guess my thought is create-react-app is 0 configuration, extremely straightforward. I want to eliminate the "choice of tool" if you're absolutely brand new to static sites.

E.g, I'm using CRA because I want to build a react app. With C-S-S, I'm using it to build a website. My initial choice becomes harder if I have to choose Jekyll, or Hugo, or Hexo, etc...

I'm thinking the most basic from of C-S-S, it provides:
- best-practise javascript w/ webpack
- best-practise css w/ postcss
- image processing
- svg sprites
- static HTML in the site/ folder to be used with your processed assets.

Thoughts?

**Comment by budparr** [comment]
I see. Bare Metal. 

> best-practise javascript w/ webpack

Are you thinking separate files for prod/dev or one? I went with one to keep it simple, but now you've got me studying CRA and am musing over having two.

**Comment by chrisdmacrae** [comment]
I went with two webpack configs because they're so heavy and I have detailed comments.

**Comment by ncphillips** [comment]
Agreed. We use one webpack configuration at Forestry right now, and I deeply regret it. The duplication is worth the simplicity.
